### PR TITLE
Deprecation/order translational

### LINKED ISSUE #659: Deprecate Translational order parameter.
I support deprecation in 2.4 and removal in 3.0. Then we could simplify the joint HexaticTranslational implementation to be “only Hexatic.”

_Originally posted by @bdice in https://github.com/glotzerlab/freud/issues/655#issuecomment-698988916_

### PR DESCRIPTION
**Pull request by vyasr**
## Description

Marked the translational OP for deprecation.

## Motivation and Context

Resolves: #659 

## How Has This Been Tested?

Nothing right now. If we start deprecating more code we can start using something like the deprecation package to manage deprecations, but at the moment I don't anticipate there being enough deprecation to merit the effort. Wait until there are three ;)

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds or improves functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation improvement (updates to user guides, docstrings, or developer docs)
- [x] Deprecation

## Checklist:

- [x] I have read the [**CONTRIBUTING**](https://github.com/glotzerlab/freud/blob/master/CONTRIBUTING.md) document.
- [x] My code follows the code style of this project.
- [x] I have updated the documentation (if relevant).
- [x] I have added tests that cover my changes (if relevant).
- [x] All new and existing tests passed.
- [ ] I have updated the [credits](https://github.com/glotzerlab/freud/blob/master/doc/source/reference/credits.rst).
- [x] I have updated the [Changelog](https://github.com/glotzerlab/freud/blob/master/ChangeLog.md).

### CODE REVIEW

File: freud/order.pyx [hunk 1]
```diff
@@ -412,13 +413,18 @@ cdef class Translational(_PairCompute):
         **2D:** :class:`freud.order.Translational` is only defined for 2D
         systems. The points must be passed in as :code:`[x, y, 0]`.
 
+    .. note::
+        This class is slated for deprecation and will be removed in freud 3.0
```
Comment (bdice) [inline freud/order.pyx]:
```suggestion
        This class is deprecated and will be removed in freud 3.0.
```

**Review by bdice — APPROVED** [review]
@vyasr If you also to put a comment in the C++ code, that would be a nice improvement. Otherwise this is good to go.

File: freud/order.pyx [hunk 2]
```diff
@@ -412,13 +413,18 @@ cdef class Translational(_PairCompute):
         **2D:** :class:`freud.order.Translational` is only defined for 2D
         systems. The points must be passed in as :code:`[x, y, 0]`.
 
+    .. note::
```
Comment (bdice) [inline freud/order.pyx:416]: Might need a newline below this for consistency with the rest of our docs -- please check.

Comment (vyasr) [reply freud/order.pyx:416]: I looked at the example directly above this one, as well as files like msd and pmft, and I don't see such a newline anywhere, so this seems fine.

**Comment by vyasr** [comment]
> @vyasr If you also to put a comment in the C++ code, that would be a nice improvement. Otherwise this is good to go.

Done

File: freud/order.pyx (hunk 2 above)
Comment (vyasr) [reply freud/order.pyx:416]: Also built and double-checked just to be sure.

**Pull request merged**